[ISSUE-2936][Feature] Support flink native type savepoint

xbthink opened the pull request:
<!--
Thank you for contributing to StreamPark! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

## Contribution Checklist

  - If this is your first time, please read our contributor guidelines: [Submit Code](https://streampark.apache.org/community/submit_guide/submit_code).

  - Make sure that the pull request corresponds to a [GITHUB issue](https://github.com/apache/incubator-streampark/issues).

  - Name the pull request in the form "[Feature] Title of the pull request", where *Feature* can be replaced by `Hotfix`, `Bug`, etc.

  - Fill out the template below to describe the changes contributed by the pull request. That will give reviewers the context they need to do the review.

  - If the PR is unfinished, add `[WIP]` in your PR title, e.g., `[WIP][Feature] Title of the pull request`.

-->

## What changes were proposed in this pull request

Issue Number: close #2936 <!-- REMOVE this line if no issue to close -->

<!--(For example: This pull request proposed to add checkstyle plugin).-->

## Brief change log

<!--*(for example:)*
- *Add maven-checkstyle-plugin to root pom.xml*
-->

## Verifying this change

<!--*(Please pick either of the following options)*-->

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

<!--*(example:)*
- *Added integration tests for end-to-end.*
- *Added *Test to verify the change.*
- *Manually verified the change by testing locally.* -->

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes / no)

xbthink committed:
support flink native type savepoint

xbthink commented:
I have used this pr at internal environment, and I am not familiar with the FE code.
Please review code, and give some suggestions. Thank you very much.

zhoulii commented on streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ApplicationServiceImpl.java:
```diff
@@ -1319,16 +1320,19 @@ public void cancel(Application appParam) throws Exception {
               }
               if (isKubernetesApp(application)) {
                 k8SFlinkTrackMonitor.unWatching(toTrackId(application));
+                log.info("k8SFlinkTrackMonitor.unWatching");
```
we can make the log more readable.

xbthink replied:
It is debug log, I have removed it.

zhoulii commented on streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ApplicationServiceImpl.java:
```diff
@@ -1319,16 +1320,19 @@ public void cancel(Application appParam) throws Exception {
             e -> {
               if (e.getCause() instanceof CancellationException) {
                 updateToStopped(application);
+                log.info("updateToStopped");
```
same as above.

zhoulii commented on streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ApplicationServiceImpl.java:
```diff
@@ -1319,16 +1320,19 @@ public void cancel(Application appParam) throws Exception {
+                log.info("updateToStopped");
               } else {
-                log.error("stop flink job fail.", e);
+                log.info("stop flink job fail.", e);
```
maybe  'log.warn' is better here.

zhoulii commented on streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/controller/AccessTokenController.java:
```diff
@@ -186,6 +186,7 @@ public RestResponse copyRestApiCurl(
               .addFormData("id", appId)
               .addFormData("savePointed", "false")
               .addFormData("drain", "false")
+              .addFormData("type", "canonical")
```
`flink/app/cancel` does not take a parameter named `type`, maybe you mean  `.addFormData("nativeFormat", "false")` here.

xbthink replied:
yes, I have fixed it.

zhoulii commented on streampark-console/streampark-console-webapp/src/views/flink/app/components/AppView/StopApplicationModal.vue:
```diff
@@ -79,6 +79,17 @@
         defaultValue: false,
         afterItem: () => h('span', { class: 'conf-switch' }, 'Send max watermark before stopped'),
       },
+      {
+        field: 'nativeFormat',
+        label: 'NativeFormat',
+        component: 'Switch',
+        componentProps: {
+          checkedChildren: 'ON',
+          unCheckedChildren: 'OFF',
+        },
+        defaultValue: false,
+        afterItem: () => h('span', { class: 'conf-switch' }, 'Use savepoint native format'),
+      },
```
It's better hide this component when canceling without savepoint, besides, we may provide more info for users, like native format is supported since flink 1.15.

zhoulii reviewed (commented):
Thanks for your contribution, @xbthink , left a few comments, PTAL.

xbthink committed:
remove some debug loggers

xbthink committed:
fix a mistake

xbthink committed:
refine code:
hide native format switch component without savepoint.
add supported flink version note of native format.

xbthink commented:
Thanks for your reviewing. I have modified codes, you can review again. @zhoulii

zhoulii reviewed (approved):
LGTM
